stock_available_to_promise_release/models/stock_move.py

```diff
@@ -51,9 +52,50 @@ class StockMove(models.Model):
         search="_search_release_ready",
     )
     need_release = fields.Boolean(index=True, copy=False)
+    unrelease_allowed = fields.Boolean(compute="_compute_unrelease_allowed")
     zip_code = fields.Char(related="partner_id.zip", store=True)
     city = fields.Char(related="partner_id.city", store=True)
 
+    @api.depends("rule_id", "rule_id.available_to_promise_defer_pull")
+    def _compute_unrelease_allowed(self):
+        user_is_allowed = self.env.user.has_group("stock.group_stock_user")
+        move_orig_ids_by_depth = self._get_all_orig_ids_by_depth(prefetch=True)
+        for move in self:
+            unrelease_allowed = (
+                user_is_allowed
+                and not move.need_release
+                and move.rule_id.available_to_promise_defer_pull
```

Comment:
and picking_type_id.code == 'outgoing' like done here https://github.com/OCA/wms/pull/478/files#diff-05db432ffe85dc4ee9d20893829c7d8c96d484cd2cb12c0be7e0a70a949be3d9R29

Reply:
:+1: